### PR TITLE
Add bbdb, elfeed, and elfeed-org

### PR DESCRIPTION
**Pull request by therockmandolinist**
I've grouped the last two under the "elfeed" folder since elfeed-org is an enhancement to elfeed.

### CODE REVIEW

**Comment by tarsius** [comment]
> tarsius approved these changes just now

No, I approved one out of three commits.

File: no-littering.el [hunk 1]
```diff
@@ -217,6 +217,10 @@ This variable has to be set before `no-littering' is loaded.")
 
     (setq ac-comphist-file                 (var "ac-comphist.el"))
     (setq anaconda-mode-installation-directory (etc "anaconda-mode/"))
+    (eval-after-load 'bbdb
```
Comment (tarsius) [inline no-littering.el]: *meta*

(1) A line comment on a single commit, using "Start a review".

File: no-littering.el [hunk 2]
```diff
@@ -217,6 +217,10 @@ This variable has to be set before `no-littering' is loaded.")
...
+      `(make-directory ,(var "bbdb/") t))
+    (setq bbdb-file                        (var "bbdb/bbdb.el"))
+    (setq bbdb-vcard-directory             (var "bbdb/bbdb-vcard/"))
```
Comment (tarsius) [inline no-littering.el]: *meta*

(2) And another, using "Add review comment".

**Comment by tarsius** [comment]
*meta*

(4) Ah, that's actually useful. Let's experiment some more.

Except that I cannot comment on the review, only individual line comments or here at the top-level.

File: no-littering.el (hunk 2 above)
Comment (tarsius) [inline no-littering.el]: *meta*

(5) Reply to a line-comment.

File: no-littering.el [hunk 3]
```diff
@@ -217,13 +217,22 @@ This variable has to be set before `no-littering' is loaded.")
...
     (eval-after-load 'bookmark+-1
       `(make-directory ,(var "bmkp/") t))
     (setq bmkp-current-bookmark-file       (var "bmkp/current-bookmark.el"))
     (setq bmkp-last-bookmark-file          (var "bmkp/last-bookmark.el"))
     (eval-after-load 'company-statistics
       `(make-directory ,(var "company/") t))
     (setq company-statistics-file          (var "company/statistics.el"))
+    (eval-after-load 'elfeed
```
Comment (tarsius) [inline no-littering.el]: *meta*

(6) Line-comment on "Files changed" and then "Start a review".

**Comment by tarsius** [comment]
*meta* (I am stopping that now, and will open a new pr for the actual changes later)

(8) The 6,7 review takes me to the overview and the 1,2,3 review takes me to the individual commit. Good.

**Comment by tarsius** [comment]
(9) What happens if I close...

**Comment by tarsius** [comment]
(10) ... and reopen?

**Comment by tarsius** [comment]
(11) Review still there. Good.

**Comment by tarsius** [comment]
(12) Going to force push, rebased but unmodified commits...

**Comment by tarsius** [comment]
(13) Actually let's make a non-review comment first...

File: no-littering.el [hunk 4]
```diff
@@ -217,13 +217,22 @@ This variable has to be set before `no-littering' is loaded.")
 
     (setq ac-comphist-file                 (var "ac-comphist.el"))
     (setq anaconda-mode-installation-directory (etc "anaconda-mode/"))
+    (eval-after-load 'bbdb
+      `(make-directory ,(var "bbdb/") t))
+    (setq bbdb-file                        (var "bbdb/bbdb.el"))
```
Comment (tarsius) [inline no-littering.el]: (14) Non-review line comment.

**Comment by tarsius** [comment]
(14) Pushing now...

**Comment by tarsius** [comment]
(15) ... or not, it seems you have disabled this.

@therockmandolinist Please rebase and then push, so that I can continue this experiment.

**Comment by tarsius** [comment]
(16) Old commits are still thrown away, but at least the diffs are kept. Sometimes that's enough...

... sometimes not. Let's see if the old mechanism for preventing that still works:

"Mentioning the current tip: 4e22b51f081381dfbc9577a066687338e3cf2af7."

File: no-littering.el [hunk 5]
```diff
@@ -234,6 +234,7 @@ This variable has to be set before `no-littering' is loaded.")
        `(make-directory ,(var "elfeed/") t))
     (setq elfeed-db-directory              (var "elfeed/db/"))
     (setq elfeed-enclosure-default-dir     (var "elfeed/enclosures/"))
+    (setq rmh-elfeed-org-files             (list (var "elfeed/elfeed.org")))
```
Comment (tarsius) [inline no-littering.el]: (17) Will this be preserved?

**Comment by tarsius** [comment]
(19) No.

File: no-littering.el (hunk 5 above)
Comment (tarsius) [inline no-littering.el]: (20) But somehow this comment got moved from 4e22b51f081381dfbc9577a066687338e3cf2af7 to dd162629c6bc3d06bd8dca9cf800a75d58475c08, which is at least surprising.

Comment (tarsius) [inline no-littering.el]: (21) What if I mention its hash when commenting on that commit? That would be dd162629c6bc3d06bd8dca9cf800a75d58475c08 at this time. Using "Add single comment".

**Comment by tarsius** [comment]
(22) Pushing...

**Comment by tarsius** [comment]
(23) Still no. Well at least the diffs are there and if we really want to be able to get our hands on the old tip, then we can still mention its hash to preserve it. However we have lost the following:

1. From the comment it is not possible to extend the diff to see the context.
2. From the commit it is not possible to see the comment.

But I guess we can open two windows.

**Comment by therockmandolinist** [comment]
Sorry, could you clarify which part you don't like? Not sure I understand. Is it simply the repetition of `bbdb` in the file name and folder? They could be renamed.

**Comment by tarsius** [comment]
> Sorry, could you clarify which part you don't like? Not sure I understand. Is it simply the repetition of `bbdb` in the file name and folder? They could be renamed.

Yes.

> [File names (2)] The name of the respective Emacs package should serve as the prefix of the file name, unless the file is in a subdirectory in which case the name of the subdirectory serves as the prefix.

Implies that the prefix should not be repeated. But if we remove that, what remains of `bbdb-file` if we also follow

> [File names (4)]  If the name of a path variable ends with -file, -default-file, -directory, -default-directory, or something similar, then that suffix is usually dropped from the file name.

`-`?!

Normally that would not be an issue because of

> [File location and subdirectories (5)] If a package only defines a single variable that specifies a data (or config) directory, then the directory name should nevertheless be just the package name.

But because of

> [File location and subdirectories (4)] A package that provides a "framework" for other packages to use, then we may reuse its directories for other packages that make use of that framework or otherwise "extend" the "main package". 

it is in this case.

**Comment by tarsius** [comment]
wip

**Comment by tarsius** [comment]
* There's no way around `bbdb/bbdb.el`, but I changed `bbdb/vcard.el` instead of `bbdb/bbdb-vcard.el`.
* I also moved the definition of `rmh-elfeed-org-files` and changed the path to`elfeed/rmh-elfeed.org`.

**Pull request closed without merging**